### PR TITLE
slcli image list should use an iterative cal

### PR DESCRIPTION
**Pull request by caberos**
[#issue1993](https://github.com/softlayer/softlayer-python/issues/1993)

### CODE REVIEW

File: SoftLayer/managers/image.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def list_private_images(self, guid=None, name=None, **kwargs):
         kwargs['filter'] = _filter.to_dict()
 
         account = self.client['Account']
-        return account.getPrivateBlockDeviceTemplateGroups(**kwargs)
+        return account.getPrivateBlockDeviceTemplateGroups(**kwargs, limit=limit)
```
Comment (allmightyspiff) [inline SoftLayer/managers/image.py:71]: Change this to `self.client.iter_call`

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

**Comment by JayasilanJibm** [comment]
Hi @caberos,

I have tested the code, seems like the image list is over 155. And with limit option as well it is different. Different output is observed each time.

Please let me know if I am doing it right or the API is returning the images in an arbitrary fashion.

Attached screenshots for further consideration.

Screenshot 2023-06-28 at 12 15 19 AM

Screenshot 2023-06-28 at 12 15 40 AM

Screenshot 2023-06-28 at 1 00 25 AM


Thanks,
Jayasilan

**Comment by caberos** [comment]
Hi @JayasilanJibm.

This command has multiples options to sent, as --public and --private options, in case not sent nothing option, this result is a sum between the public and private images. 

image

Thanks
Caberos

**Comment by JayasilanJibm** [comment]
Hi,

I have tested the code changes. The same result has been observed with limit option as well. 

Approving this PR.

Please find the screenshot for further consideration.

Screenshot 2023-06-28 at 1 19 54 PM

Screenshot 2023-06-28 at 1 20 04 PM



Thanks,
Jayasilan

**Review by JayasilanJibm — APPROVED** [review]

File: SoftLayer/managers/image.py (hunk 1 above)
Comment (ramkishor-ch) [reply SoftLayer/managers/image.py:71]: Hi @caberos,

I have doubts.

I saw that you have added code **self.client.iter_call** in **def list_public_images()** function but **not** **added** in **def list_private_images()** function.

**Doubts:**
1. Is it possible to add self.client.iter_call in def list_private_images() function?
   a. If yes, can you please add it,
   b. If no, can you please share the reason behind it.?

2. I see that 
File path: softlayer-python/SoftLayer/fixtures/SoftLayer_Virtual_Guest_Block_Device_Template_Group.py

There are no fixtures of related to private images.

Thanks,
Ramkishor Chaladi.

Comment (allmightyspiff) [inline SoftLayer/managers/image.py:71]:
```suggestion
        return return self.client.iter_call('Virtual_Guest_Block_Device_Template_Group' , 'getPrivateBlockDeviceTemplateGroups', **kwargs, limit=limit)
```

File: SoftLayer/managers/image.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def list_public_images(self, guid=None, name=None, limit=100, **kwargs):
 
         kwargs['filter'] = _filter.to_dict()
 
-        return self.vgbdtg.getPublicImages(**kwargs, limit=limit)
+        return self.client.call('Virtual_Guest_Block_Device_Template_Group', 'getPublicImages', **kwargs, limit=limit)
```
Comment (allmightyspiff) [inline SoftLayer/managers/image.py:92]:
```suggestion
        return self.client.iter_call('Virtual_Guest_Block_Device_Template_Group', 'getPublicImages', **kwargs, limit=limit)
```

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

**Pull request merged**